### ISSUE TITLE
Pimax eye tracking module support

### ISSUE THREAD

**Issue by RileyGuy** [Frooxius/NeosPublic#1696]
## Is your feature request related to a problem? Please describe.
No.

## Relevant issues
N/A

## Describe the solution you'd like
Support could be added for the Pimax 8k with the eye tracking module.

## Describe alternatives you've considered
None that I can think of, this seems like something that would ideally be added to the code base.

## Additional context
I've run into a user (infotron) who has a Pimax 8k with the eye tracking module and they report that it doesn't work with neos. Since they don't know how to use Github, I figured I would make a post in their stead.

**Comment by shiftyscales** [comment]
#1140 could be seen as a relevant issue.

From what I'm reading the Droolon Pi1 should be utilizing SRanipal like the Vive Pro Eye? If it isn't, do you have a specific link to the SDK/implementation details on how to support that device?

Also, has the user tried using the `-ForceSRanipal` command line argument to try to force the use of SRanipal? @RileyGuy

**Comment by RileyGuy** [comment]
They've tried all of the tricks, tried the argument and even tried special reverse-engineered drivers to force it through SRanipal to no avail.

**Comment by InsaneGrox** [comment]
from what I can tell, pimax's eye tracker does NOT use sranipal, and instead uses a piece of software called aSeeVR, the SDK apparently is a pain to get your hands on however, closest I could find is this guide from pimax that says the SDK is in the install directory? https://www.pimax.com/pages/eye-tracking

**Comment by micmac253** [comment]
if it helps here is the aseevr user sdk. https://drive.google.com/file/d/1cyqap5TOA5lb8A82o3jGlDWXamrdS5U-/view?usp=sharing

**Comment by Frooxius** [comment]
Do you have an official link to the SDK? The one on Pimax.com gives 404 error and I'd rather avoid using unknown sources like GDrive. If it's just an SDK issue this shouldn't be difficult to get integrated, but I'm quite sure where to get it officially.

**Comment by micmac253** [comment]
> Do you have an official link to the SDK? The one on Pimax.com gives 404 error and I'd rather avoid using unknown sources like GDrive. If it's just an SDK issue this shouldn't be difficult to get integrated, but I'm quite sure where to get it officially.

I cant seem to find a download for it on the site. would you prefer another way of me sending it? thats my google drive I uploaded it to.

**Comment by Frooxius** [comment]
It's not about method of sending it, but about using a proper source. I need to get it from an official source for integration for safety and legal reasons and future updates.

**Comment by micmac253** [comment]
> It's not about method of sending it, but about using a proper source. I need to get it from an official source for integration for safety and legal reasons and future updates.

Ah I see, I will contact pimax about the sdk having a 404 error.

**Comment by micmac253** [comment]
> It's not about method of sending it, but about using a proper source. I need to get it from an official source for integration for safety and legal reasons and future updates.

I have found an official download link from the manufacturer of the eye tracking module for the sdk. Is this ok? https://www.7invensun.com/filedownload/228933

**Comment by micmac253** [comment]
https://pitool.pimax.com/aSeeVR.exe here is a link to download the actual executable file itself if that helps

**Comment by kazu0617** [comment]
> 
> 
> > It's not about method of sending it, but about using a proper source. I need to get it from an official source for integration for safety and legal reasons and future updates.
> 
> I have found an official download link from the manufacturer of the eye tracking module for the sdk. Is this ok? https://www.7invensun.com/filedownload/228933

I think this one are more formally and convenient. https://www.7invensun.com/aglassgbbsdkjdemo

And here is a product page: https://www.7invensun.com/droolonpi1xqy

**Comment by dfgHiatus** [comment]
Seeing how this hasn't been posted here, I'll post it: https://github.com/NGenesis/PimaxEyeTracker

Understandably lack of proper SDK poses a handful of problems. That said, this may be a viable alternative.

MIT licensed and exposes a C# API.

**Comment by Xlinka** [comment]
i'm surprised no one linked the actual Pimax developer page. to get the unity SDK https://pimax.com/developer/
![image](https://user-images.githubusercontent.com/22996716/141853539-f48365ef-3bd1-45f1-baa5-fc8d7f1b58d4.png)
click to download default install is to C:/7invensun/ and will have a folder called doc's where everything is documented all the files and classes 
![image](https://user-images.githubusercontent.com/22996716/141853653-34b99167-30bf-4abd-919d-1d0b4e2b3eb7.png)
this also includes a c++ example on how to use the eye tracker.

**Comment by shiftyscales** [comment]
I'm fairly certain that page just quite literally did not exist prior, @Xlinka.

**Comment by Xlinka** [comment]
@shiftyscales yeah originally was intended for business partners and developers they invited i think around august the page became public before it would 404 or 405

**Comment by micmac253** [comment]
Yeah there was no developer page at all, I even contacted Pimax and all they said was the sdk would download automatically with pitool, meaning no actual dev support.

**Comment by micmac253** [comment]
But now that those tools and resources are available, it would be nice to finally have this implemented. I'm really tired of using my vive lol

**Comment by dfgHiatus** [comment]
I have a workaround for this issue: https://github.com/dfgHiatus/NeosPimaxEyeTracking
